doc/manual/source/nethsm.rst

```diff
@@ -0,0 +1,464 @@
+will be wiped when the container is stopped.
+
+.. code-block:: bash
+
+   $ docker run --rm -ti -p 127.0.0.1:8443:8443 docker.io/nitrokey/nethsm:testing
+
+Install the Prerequisites
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   $ sudo apt install -y opensc opensc-pkcs11 pipx
+   $ pipx install pynitrokey    # for the nitropy utility
+   $ pipx ensurepath
+   $ source ~/.bashrc
+   $ export NETPKCS="/usr/lib/x86_64-linux-gnu/nethsm-pkcs11.so"
+
+We download and install the NetHSM PKCS#11 driver to a somewhat shorter name;
+``$NETPKCS`` will help us keep command-lines short in this documentation.
+
+.. code-block:: bash
+
+    $ wget https://github.com/Nitrokey/nethsm-pkcs11/releases/download/v1.7.2/nethsm-pkcs11-v1.7.2-x86_64-linux-glibc.so
+    $ sudo install nethsm-pkcs11-v1.7.2-x86_64-linux-glibc.so $NETPKCS
+
+Configure the NetHSM PKCS#11 driver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We then configure the driver, in the file ``/etc/nitrokey/p11nethsm.conf``,
+paying attention to the URI, usernames, and passwords.  as per `NetHSM's
+PKCS#11 Setup documentation <https://docs.nitrokey.com/nethsm/pkcs11-setup>`_
+
+.. code-block:: text
+
+    enable_set_attribute_value: false
+    log_level: Debug
+    syslog_facility: "user"
+
+    slots:
+      - label: LocalHSM                        # Name your NetHSM however you want
+        description: Local HSM (docker)        # Optional description
+        operator:
+          username: "jj01"
+          password: "blab123456"
+        administrator:
+          username: "admin18"
+          password: "secret9999"
+
+        instances:
+          - url: "https://127.0.0.1:8443/api/v1"   # URL to reach the server
+            max_idle_connections: 10
+            danger_insecure_cert: true
+        retries:
+          count: 3
+          delay_seconds: 1
+        tcp_keepalive:
+          time_seconds: 600
+          interval_seconds: 60
+          retries: 3
+        connections_max_idle_duration: 1800
+        timeout_seconds: 10
+
+Provision the HSM
+~~~~~~~~~~~~~~~~~
+
+Configure access to the NetHSM, ensuring IP address and port number match those
+of the container running the HSM. We show passwords in clear below so as to be
+able to demonstrate where they are later used. Please and obviously don't use
+these. As we haven't `configured a TLS key and certificate for the device
+<https://docs.nitrokey.com/nethsm/administration#tls-certificate>`_ we
+disable TLS verification on the connection (not recommended).
+
+.. code-block:: bash
+
+   $ export NETHSM_HOST=127.0.0.1:8443
+
+   $ nitropy nethsm --no-verify-tls provision
+   Command line tool to interact with Nitrokey devices 0.11.2
+   Unlock passphrase: nlnetlabs001
+   Repeat for confirmation:
+   Admin passphrase: lecascadeur
+   Repeat for confirmation:
+   Warning: The unlock passphrase cannot be reset without knowing the current value...
+   NetHSM 127.0.0.1:8443 provisioned
+
+
+Configure NetHSM's TLS certificate
```

Comment:
imo this entire section conflicts with the use of `danger_insecure_cert: true` PKCS#11 driver and the assumption that the HSM container is run locally. personally I'd ditch this and stick to `--no-verify-tls` with the nitropy tool. Maybe with a short reference link to the Nitrokey documentation for how this is done right in a prod setup.

Reply:
I hadn't yet got around to actually testing `danger_insecure_cert: false`, but you are right in that.

I documented the effort because I couldn't actually find documentation for it and under the _assumption_ that people wouldn't typically be using the OCI image as an HSM ...

Reply:
`danger_insecure_cert: false` works as advertised.

Reply:
to be honest, I've also started documenting the nitrokey integration after my findings in https://github.com/NLnetLabs/kmip2pkcs11/issues/14 but It didn't meet my requirements yet for having this somewhere public, the commands haven't been tested front to back to be all copy/pastable etc ... However, as you are pretty much writing the same thing, feel free to compare and rip anything from my mumblings https://gist.github.com/and0x000/a50bdb2f883247bf00426c1c1140c091

It's a pretty shameless rewrite of the thales documentation anyway 🙈